src/auth/mod.ts

```diff
@@ -15,33 +14,131 @@ interface DecodedToken {
 
 const logger = getLogger(import.meta);
 
-function isSubInPodsConfig(payload: DecodedToken): boolean {
-  return podsConfig.pods.some((pod) =>
-    pod.sub === payload.sub &&
-    pod.serviceaccount.uid === payload["kubernetes.io"]?.serviceaccount.uid
+export async function authenicateRequestAndReturnPodServiceAccount(
+  token: string,
+): Promise<string> {
+  const payload = await verifyToken(token);
+
+  if (!payload["kubernetes.io"]) {
+    logger.error("Payload does not contain kubernetes.io field");
+    throw new Error("Payload does not contain kubernetes.io field");
+  }
+
+  return payload["kubernetes.io"].serviceaccount.name;
+}
+
+async function verifyToken(token: string): Promise<DecodedToken> {
+  const key = await getPublicKey();
```

Comment:
this should be happening only one at the start of the app (recall the discussion about not overloading the apiserver)

Reply:
yeah, will be adding this, didn't forget :) will fetch public keys periodically and store them in a env var, that's what I have planned.